src/components/forms/fields/InputField/InputField.tsx

```diff
@@ -55,6 +59,41 @@ export const InputField = (props: InputFieldProps) => {
     unstyled: tags && tags.length > 0,
   };
 
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    // first handle supplied onChange, if exists
+    if (inputProps.onChange) {
+      inputProps.onChange(e);
+    }
+    // then return value to onUpdate
+    if (onUpdate) {
+      onUpdate(e.target.value);
+    }
+  };
+
+  const onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    // first handle supplied onKeyUp, if exists
+    if (inputProps.onKeyUp) {
+      inputProps.onKeyUp(e);
+    }
+    // then return value to onUpdateTags
+    if (onUpdateTags && onUpdate) {
+      const { value } = inputProps;
+      if (e.key === 'Backspace' && value === '') {
+        onUpdateTags(tags.slice(0,-1));
+      }
+      if (e.key === 'Enter' && value !== '') {
+        onUpdateTags([...tags as string[], value as string]);
```

Comment:
Are the casts necessary here? They seem unsafe. If it's to get rid of an `undefined` case can do something like `[...(tags || [])]`. I'm looking at this on GitHub so can't see the inferred types at this point.